fix(ui): ERC20 burn fails if amount burned > holdings.

asishap opened the pull request:
# Description

Fixed the issue when ERC20 burn fails if amount burned greater than holdings.

## Related Issue
## Motivation and Context

UI gets hangs and throwing contract error when user enters a higher value than his holdings on ERC-20 token burn.  Fixed by checking the enter value is greater that he have and shows a toast message in UI. 

## How Has This Been Tested

Tested using UI

## Screenshots (if appropriate)

## Types of changes


- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

asishap commented:
> Tested in the UI with no issues for me. What would you think about propagating these changes to other situations where the user has insufficient tokens?

@ocgordon ,
For token commitments we already showing the error messages if user doesn't have token commitments. For ERC20 token transfer/burn user need to select token from select input. So if user does't have ERC20 tokens then it wont be listed .

iAmMichaelConnor commented:
@asishap I'm confused about what this PR is addressing. In the 'Motivation and Context' section, it says the issue relates to a *mint*. But the PR addresses *burning*. Please can you update the details to be clear?

Please can you also solve the current merge conflicts (e.g. by merging master into this branch)?

asishap commented:
> @asishap I'm confused about what this PR is addressing. In the 'Motivation and Context' section, it says the issue relates to a _mint_. But the PR addresses _burning_. Please can you update the details to be clear?
> 
> Please can you also solve the current merge conflicts (e.g. by merging master into this branch)?

@iAmMichaelConnor 

I have corrected the 'Motivation and Context' section and updated the PR. Please take a look.
